dev/ci/gitlab-ci-ctests.yml

```
@@ -26,14 +26,22 @@
   extends: .base_config
   stage: create_experiments
   script: |
-    set -e  # Fail the job if any command fails
     echo "Setting up for ctests workflow"
-    source ${GW_HOMEgfs}/dev/ci/platforms/config.${machine}
-    cd ${GW_HOMEgfs}/dev/ctests
+    echo "Using build directory ${GW_HOMEgfs}"
+    source "${GW_HOMEgfs}/dev/ush/gw_setup.sh"
+    source "${GW_HOMEgfs}/dev/ci/platforms/config.${machine}"
+    cd "${GW_HOMEgfs}/dev/ctests" || exit 1
     mkdir -p build
-    cd build
-    cmake -S ${GW_HOMEgfs}
+    cd build || exit 1
+    cmake -S "${GW_HOMEgfs}"
     ctest -N
+    ctest_status=$?
+    if [ "${ctest_status}" -ne 0 ]; then
+      echo "==========================================================="
+      echo "ERROR: CMake configuration failed with exit code ${ctest_status}"
+      echo "==========================================================="
+      exit 1  # Force exit code 1 for GitLab to recognize the failure
+    fi
     num=$(ctest -N | grep "Total Tests" | awk "{print \$3}")
     [ "$num" -gt 0 ] || { echo "No tests found"; exit 1; }
```

`[[ ... ]] || do some thing` logic is prone to unexpected errors.  Suggest making this into an if block.
```suggestion
    if [[ ${num} -le 0 ]; then
      echo "No tests found"
      exit 1
    fi